io_buffer.c

```diff
@@ -708,31 +715,48 @@ io_buffer_map(int argc, VALUE *argv, VALUE klass)
     // We might like to handle a string path?
     VALUE io = argv[0];
 
+    rb_off_t file_size = rb_file_size(io);
+    // Compiler can confirm that we handled file_size <= 0 case:
+    if (UNLIKELY(file_size <= 0)) {
```

Comment:
```suggestion
    if (UNLIKELY(file_size < 0)) {
```

In wonder if we should allow mapping zero-sized files? Not sure what the point would be, but is it technically invalid?

Reply:
My goal is to avoid imposing additional semantics. However, as you have pointed out, it's pretty tricky to find the union of "semantics" across a wide range of platforms without making it the user's problem.